jobs/build/ocp4/build.groovy

```diff
@@ -0,0 +1,595 @@
+    if (imageList) {
+        buildPlan[imageList] = commonlib.cleanCommaList(params.IMAGE_LIST)
+    } else if (params.IMAGE_LIST.trim()) {
+        error("aborting because a list of images was specified; you probably want to specify only/except.")
+    }
+
+    echo "Initial build plan: ${buildPlan}"
+
+    // figure out the puddle configuration
+    aosCdJobsCommitSha = commonlib.shell(script: "git rev-parse HEAD", returnStdout: true).trim()
+    def puddleConfBase = "https://raw.githubusercontent.com/openshift/aos-cd-jobs/" +
+                         "${aosCdJobsCommitSha}/build-scripts/puddle-conf"
+    buildPlan.puddleConf = "${puddleConfBase}/atomic_openshift-${version.stream}.conf"
+    // and where to mirror the compose when it's done
+    rpmMirror << [
+        target: params.RPM_MIRROR ?: "pre-release",
+        textTarget: (params.RPM_MIRROR == "online:int") ? "(Integration Testing)" : "(Release Candidate)",
+        url: getMirrorUrl(params.RPM_MIRROR, version.stream),
+    ]
+
+    // adjust the build "title"
+    currentBuild.displayName = "#${currentBuild.number} - ${version.full}-${version.release}"
+    if (buildPlan.dryRun) { currentBuild.displayName += " [DRY RUN]" }
+    if (buildPlan.forceBuild) { currentBuild.displayName += " [force build]" }
+    if (!buildPlan.buildRpms) { currentBuild.displayName += " [no RPMs]" }
+    if (!buildPlan.buildImages) { currentBuild.displayName += " [no images]" }
+
+    // get a fresh doozerWorking; removing the old one is left in the background.
+    // NOTE: this waits for the background process if wrapped in commonlib.shell;
+    // as this is designed to run instantly and never fail, just run it in a normal shell.
+    sh """
+        mkdir -p ${doozerWorking}
+        mv ${doozerWorking} ${doozerWorking}.rm.${currentBuild.number}
+        mkdir -p ${doozerWorking}
+        # see discussion at https://stackoverflow.com/a/37161006 re:
+        JENKINS_NODE_COOKIE=dontKill BUILD_ID=dontKill nohup bash -c 'rm -rf ${doozerWorking}.rm.*' &
+    """
+    return planBuilds()
+}
+
+
+// From a brew NVR of openshift, return just the V part.
+@NonCPS
+def extractBuildVersion(build) {
+    // closure also keeps regex away from pipeline steps (error|echo)
+    def match = build =~ /(?x) ^openshift- (  \d+  ( \. \d+ )+  )-/
+    return match ? match[0][1] : "" // first group in the regex
+}
+
+/**
+ * From the minor version (stream) and parameters, determine which version to build.
+ * @param stream: OCP minor version "X.Y"
+ * @return a map to merge into the "version" property representing the determined version
+ */
+def determineBuildVersion(stream) {
+    def full = "${stream}.0"  // default
+    def release = new Date().format("yyyyMMddHHmm")
+
+    def prevBuild = retry(3) {
+        commonlib.shell(
+            script: "brew latest-build --quiet rhaos-${stream}-rhel-7-candidate openshift | awk '{print \$1}'",
```

Comment:
Yes, I think that is a great idea. Since doozer already manages ocp-build-data it seems natural for the job to be able to query doozer for group data, rather than have the job read it directly. Especially since this is templated. Do you want to work on providing that from doozer?

This can remain hardcoded for now and be improved at our leisure.

Reply:
Yep -> https://jira.coreos.com/browse/ART-535